common/template_tests.json

```diff
@@ -591,6 +591,13 @@
     "output": "03-10-2014",
     "errors": []
   },
+  {
+    "template": "@(TO_EPOCH(d1)) @(TO_EPOCH(d2)) @(TO_EPOCH(d2) / 1000000000)",
+    "context": {"variables": {"d1": "2017-06-12T16:56:59.123456Z", "d2": "2017-06-12T18:56:59.000000+02:00"}, "timezone": "UTC", "date_style": "day_first"},
+    "url_encode": false,
+    "output": "1497286619123456000 1497286619000000000 1497286619",
```

Comment:
The word epoch means seconds since 1970 to me. To get nano precision, I think I prefer the python approach of returning fractional seconds instead which maintains the spirit of epoch. Also a decent chance that endpoints expecting unix time will just work without the user needing to use floor.

Reply:
inclined to agree

Reply:
Ya, epoch implies seconds. Could return fractional seconds and make people `floor` or whatnot. Technically people can calculate epoch as an expression right now btw: `@(DATE_DIFF(NOW(), "1970-01-01 00:00:00Z", "s")` or something like that. :)